download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/cache/FileIndex.java

```diff
@@ -0,0 +1,127 @@
+package com.googlecode.download.maven.plugin.internal.cache;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutput;
+import java.io.ObjectOutputStream;
+import java.io.RandomAccessFile;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Binary file backed index.
+ * @author Paul Polishchuk
+ * @since 1.3.1
+ */
+final class FileIndex implements Index {
```

Comment:
@mickaelistria For example, mock implementation to simplify `DownloadCache` testing
Potentially, we could store the index in many formats, but the main reason for me now is testability.